src/mrpro/operators/LinearOperator.py

```diff
@@ -47,6 +49,99 @@ def H(self) -> LinearOperator:  # noqa: N802
+            norm and returns a tensor of shape (4,30,1) corresponding to the operator norms of the respective
+            matrices in the diagonal of the block-diagonal operator (if considered in matrix representation).
+            In any case, the output of the algorithm has the same number of dimensions as the elements of the
+            domain of the considered operator (whose dimensionality is implicitly defined by choosing dim), such that
+            the pointwise multiplication of the operator norm and elements of the domain (to be for example used
+            in a Landweber iteration) is well-defined.
+        max_iterations
+            maximum number of iterations
+        relative_tolerance
+            absolute tolerance for the change of the operator-norm at each iteration;
+            if set to zero, the maximal number of iterations is the only stopping criterion used to stop
+            the power iteration
+        absolute_tolerance
+            absolute tolerance for the change of the operator-norm at each iteration;
+            if set to zero, the maximal number of iterations is the only stopping criterion used to stop
+            the power iteration
+        callback
+            user-provided function to be called at each iteration
+
+        Returns
+        -------
+            an estimaton of the operator norm
+        """
+        if max_iterations < 1:
+            raise ValueError('The number of iterations should be larger than zero.')
+
+        norm_initial_value = torch.linalg.vector_norm(initial_value, dim=dim, keepdim=True)
+        if not (norm_initial_value > 0).all():
+            if dim is None:
+                raise ValueError('The initial value for the iteration should be different from the zero-vector.')
+            else:
+                raise ValueError(
+                    'Found at least one zero-vector as starting point. \
+                    For each of the considered operators, the initial value for the iteration \
+                    should be different from the zero-vector.'
+                )
+
+        vector = initial_value
+
+        # creaty dummy operator norm value that cannot be correct because by definition, the
+        # operator norm is a strictly positive number. This ensures that the first time the
+        # change between the old and the new estimate of the operator norm is non-zero and
+        # thus prevents the loop from exiting despite a non-correct estimate.
+        op_norm_old = torch.zeros(*tuple([1 for _ in range(vector.ndim)]))
+
+        for _ in range(max_iterations):
+            # apply the operator to the vector
+            (vector_new,) = self.adjoint(*self(vector))
+
+            product = vector.real * vector_new.real
+            if vector.is_complex() and vector_new.is_complex():
+                product += vector.imag * vector_new.imag
+            dim = tuple(dim) if dim is not None else dim
```

Comment:
move this out of the loop

Reply:
what do you mean by that?  if we move it out of the loop, we only compute it once, which does not allow to track the operator-norm. 
do you mean to move out the code and define some function `def inner_product(vector1, vector2)` and to call that inside the loop?

Reply:
or do you mean only the part `dim = tuple(dim) if dim is not None else dim` ?

Reply:
the latter :)